OpenQA/Isotovideo/Utils.pm

```diff
@@ -14,13 +14,13 @@ use Try::Tiny;
 our @EXPORT_OK = qw(git_rev_parse checkout_git_repo_and_branch
   checkout_git_refspec handle_generated_assets load_test_schedule);
 
-sub git_rev_parse ($dirname) {
-    return 'UNKNOWN' unless -e "$dirname/.git";
+sub git_rev_parse ($dirname, $cmd_prefix='') {
     $dirname = path($dirname)->realpath;
-    my $checksafe = q{git config --global --get safe.directory | grep -q};
-    my $addsafe = q{HOME=$(mktemp -d --tmpdir os-autoinst-git.XXXXX) && git config --global --add safe.directory};
-    my $version = qx{($checksafe "$dirname" && git -C "$dirname" rev-parse HEAD || $addsafe "$dirname" && git -C "$dirname" rev-parse HEAD && rm -r \$HOME)};
-    $version ||= '(unreadable git hash)';
+    chomp(my $version = qx{$cmd_prefix git -C "$dirname" rev-parse HEAD 2>&1});
+    return $version if $? == 0;
+    return 'UNKNOWN' unless $version =~ /(git config.*safe.directory.*$)/;
```

Comment:
Hm, so now we'll get "UNKNOWN" for every error (except the unsafe dir error).
I don't know why you removed the first line.

Reply:
I guess we can live with `UNKNOWN` in case of an error.

Reply:
ok